From cb7357696e138c791569ab438c8c36749147e18b Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Mon, 31 Aug 2015 11:39:15 +0800
Subject: [PATCH] MLK-11456-01 tty: serial: imx: add pinctrl sleep mode
 support

For LPSR mode, UART iomux settings will be lost after resume,
so add pinctrl sleep mode support.

Signed-off-by: Anson Huang <b20788@freescale.com>
Signed-off-by: Fugang Duan <B38611@freescale.com>
---
 drivers/tty/serial/imx.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index 37050dc..67b1526 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -1974,6 +1974,8 @@ static int serial_imx_suspend(struct platform_device *dev, pm_message_t state)
 	sport->saved_reg[9] = readl(sport->port.membase + IMX21_UTS);
 	clk_disable_unprepare(sport->clk_ipg);
 
+	pinctrl_pm_select_sleep_state(&dev->dev);
+
 	return 0;
 }
 
@@ -1982,6 +1984,8 @@ static int serial_imx_resume(struct platform_device *dev)
 	struct imx_port *sport = platform_get_drvdata(dev);
 	unsigned int val;
 
+	pinctrl_pm_select_default_state(&dev->dev);
+
 	clk_prepare_enable(sport->clk_ipg);
 	writel(sport->saved_reg[4], sport->port.membase + UFCR);
 	writel(sport->saved_reg[5], sport->port.membase + UESC);
-- 
1.8.0

